blockchain/vm/eips.go

```diff
@@ -188,3 +191,20 @@ func enable4399(jt *JumpTable) {
 		computationCost: params.RandomComputationCost,
 	}
 }
+
+// enable3855 applies EIP-3855 (PUSH0 opcode)
+func enable3855(jt *JumpTable) {
+	jt[PUSH0] = &operation{
+		execute:         opPush0,
+		constantGas:     GasQuickStep,
+		minStack:        minStack(0, 1),
+		maxStack:        maxStack(0, 1),
+		computationCost: params.PushComputationCost,
```

Comment:
I think the introduced function's running time is much faster and cheaper. Gas cost can be saved to utilize this instruction, but not affected to the computation cost in this change. They should have correleation each other. For example, if the gas is consumed bigger, then the computation cost also should be increased, and vice versa.

Reply:
@hyunsooda Good point. It is said the performance is improved a lot. You know, we usually use [this benchmark test](https://github.com/klaytn/klaytn/blob/46b075f5a83578cef74cf3d6015c64f213a23175/blockchain/vm/instructions_test.go#L1011-L1013) to decide computation cost. However, the result seems not improved. The result varies a lot.
```
➜  vm git:(PUSH0) ✗ go test -bench=BenchmarkOpPush0 -benchtime 200x
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpPush0-10          200                30.83 ns/op

➜  vm git:(PUSH0) ✗ go test -bench=BenchmarkOpPush1 -benchtime 200x
goos: darwin
goarch: arm64
pkg: github.com/klaytn/klaytn/blockchain/vm
BenchmarkOpPush1-10                  200                41.25 ns/op 
```

Moreover, i just use PUSH Computation cost because there's an issue that our computation cost measuring way is not exact.

What do you think about it?

Reply:
When comparing the actual implementation code, there is a notable difference between the opcode `PUSH0` and the `makePush` function. The `PUSH0` opcode consists of a single line that initializes the big integer type, while the `makePush` function comprises multiple instructions.

For instance, let's assume there are 100 occurrences of the `PUSH0` opcode in a contract. In terms of computational cost (CC), these `PUSH0` instructions are considered to have the same cost as other `PUSHX` instructions. However, it's important to note that the CC value is not saved, which can potentially result in transaction failures although the actual CC of `PUSH0` is cheaper. This discrepancy in CC calculation can lead to unintended issues.

Furthermore, it's essential to consider the lifetime of the contract. The contract remains active unless its code is destroyed using the `selfdestruct`. Therefore, the aforementioned computational cost must be paid continuously throughout the contract's existence. This is my overall concerns.

In summary:
- Decreased gas cost should be accompanid the decreased CC as well.
- Consider the contract lifetime and running the same multiple instructions.

Conclusion:
I believe the appropriate solution for this particular PR would be to reduce the CC associated with the `PUSH0` opcode. While addressing the removal of CC and improving its measurement is a separate topic that can be discussed at a later time.

Please feel free to add further comment.

Reply:
@hyunsooda Okay, I'll reduce the computation cost to 2/3 level of the other PUSH computation cost value. (120 -> 80) https://github.com/klaytn/klaytn/pull/1861/commits/1cc1ab0703502924ba4b318f603b8836b9e6f299.